controllers/clustergroupupgrade_controller.go

```diff
@@ -1630,18 +1690,22 @@ func (r *ClusterGroupUpgradeReconciler) updateStatus(ctx context.Context, cluste
 		}
 		policiesStatus := make([]string, 0)
 		for _, policy := range copiedPolicies.Items {
-			policiesStatus = append(policiesStatus, policy.GetName())
+			policiesStatus, err = r.checkDuplicateChildResources(ctx, clusterGroupUpgrade.Status.SafeResourceNames, policiesStatus, &policy)
+			if err != nil {
+				return err
+			}
 		}
 		clusterGroupUpgrade.Status.CopiedPolicies = policiesStatus
 
 		err = r.Status().Update(ctx, clusterGroupUpgrade)
-
 		return err
 	})
 
 	if err != nil {
 		return err
 	}
+	// Wait a bit so that API server/etcd syncs up and the next reconsile call will get the updated CGU
+	time.Sleep(statusUpdateWaitInMilliSeconds * time.Millisecond)
```

Comment:
Without this, we almost always get another reconcile call on a stale version of CGU right after one set of policies just get copied. We end up with two sets of copied policies and will rely on the name annotation based duplicate check for cleaning them up. Another side effect is that any reconcile call on a stale object will get flagged and requeued right away, which often cause another reconcile call on stale cgu again as we update status everytime. It can go on for several times until we are lucky enough to hit etcd node that just gets updated in the previous call. A bit wasteful in general and logs will look confusing. I can remove it if it really impacts our scale objective.